runtime/vm/JFRChunkWriter.cpp

```diff
@@ -1367,4 +1367,35 @@ VM_JFRChunkWriter::writeSystemProcessEvent(void *anElement, void *userData)
 	entry->commandLine = NULL;
 }
 
+void
+VM_JFRChunkWriter::writeSystemGCEvent(void *anElement, void *userData)
+{
+	SystemGCEntry *entry = (SystemGCEntry *)anElement;
+	VM_BufferWriter *bufferWriter = (VM_BufferWriter *)userData;
+
+	/* Reserve size field */
+	U_8 *dataStart = reserveEventSize(bufferWriter);
+
+	/* Write event type */
+	bufferWriter->writeLEB128(SystemGCID);
+
+	/* Write start time */
+	bufferWriter->writeLEB128(entry->ticks);
+
+	/* Write duration time which is always in ticks, in our case nanos */
+	bufferWriter->writeLEB128(entry->duration);
```

Comment:
I didnt realize there was a duration field. The version of the event I looked at didnt have one. 

In this case you need a `J9HOOK_SYSTEM_GC_CALLED` instead, which is triggered after the GC so that you know the duration of the GC.